### PR TITLE
Fix error message for footernote references with EOL

### PR DESCRIPTION
**Pull request by parhamsaremi**
When we have a commit with EOL after "[]" then we get a confusing error about the reference not existing. In this PR I first check if a reference number has no reference and give a different error about it.

### CODE REVIEW

File: commitlint/plugins.test.ts [hunk 1]
```diff
@@ -476,6 +476,17 @@ test("footer-references-existence4", () => {
     expect(footerReferenceExistence4.status).toBe(0);
 });
 
+test("footer-references-with-EOL", () => {
+    let commmitMsgwithEOLFooter =
+        "foo: this is only a title" +
+        "\n\n" +
+        "Bla bla blah[1]" +
```
Comment (knocte) [inline commitlint/plugins.test.ts:483]: @parhamsaremi this commit msg will make the paragraph rule fail

File: commitlint/helpers.ts [hunk 2]
```diff
@@ -98,6 +98,11 @@ export abstract class Helpers {
         return isLowerCase && !isUpperCase;
     }
 
+    public static isEmptyFooterReference(line: string) {
+        Helpers.assertLine(line);
+        return line[0] === "[" && line.indexOf("]") > 0 && line.trim().length === 3;
```
Comment (knocte) [inline commitlint/helpers.ts:103]: @parhamsaremi hardcoding 3 here is still wrong, not sure if you understood my previous comment

Comment (parhamsaremi) [reply commitlint/helpers.ts:103]: hmmm right. I understand now. I'll check with regex to see if it's only footer ref number.

Comment (knocte) [reply commitlint/helpers.ts:103]: I don't think using regex here is worth it, just check if the string last char before EOL (trimmed) is `]`

Comment (parhamsaremi) [reply commitlint/helpers.ts:103]: :+1:

File: commitlint/plugins.test.ts (hunk 1 above)
Comment (parhamsaremi) [reply commitlint/plugins.test.ts]: I added a . after Bla bla.

File: commitlint/helpers.ts (hunk 2 above)
Comment (parhamsaremi) [reply commitlint/helpers.ts]: In the end I just check if the indexof "]" is length-1

Comment (parhamsaremi) [reply commitlint/helpers.ts]: I think I have to use regex to catch this :thinking:  do you agree? @knocte

Comment (parhamsaremi) [reply commitlint/helpers.ts]: I changed it to regex and also there was some sanity check errors that I addressed. let me know if you still think I shouldn't use regex and checking last index to be "]" is ok.

Comment (knocte) [reply commitlint/helpers.ts]: > Although now that I think about it what if the user has something like this?

Such an edge case doesn't deserve the complexity of the code that a regex brings. If we find an edge case like that we will develop a PR on its own with its own unit test.

Comment (knocte) [reply commitlint/helpers.ts]: There are thousands or millions of bugs lurking in any piece of software, you cannot expect all of them in just one line of code. We work on a 1 by 1 basis.

Comment (parhamsaremi) [reply commitlint/helpers.ts]: makes sense. I'll bring it back

**Comment by knocte** [comment]
@aarani please review

File: commitlint/plugins.test.ts [hunk 3]
```diff
@@ -476,6 +476,19 @@ test("footer-references-existence4", () => {
...
 
+test("footer-references-with-EOL", () => {
+    let commmitMsgwithEOLFooter =
+        "foo: this is only a title" +
+        "\n\n" +
+        "Bla bla blah[1]." +
+        "\n\n" +
+        "[1]\nhttp://foo.bar/baz";
+    let footerReferenceWithEOL = runCommitLintOnMsg(commmitMsgwithEOLFooter);
+    expect(footerReferenceWithEOL.output[1].toString().includes("EOL")).toBe(
```
Comment (aarani) [inline commitlint/plugins.test.ts:487]: This expectation looks kinda weird

Comment (aarani) [reply commitlint/plugins.test.ts:487]: Isn't a better way to "expect" this?

Comment (aarani) [reply commitlint/plugins.test.ts:487]: What if we change the message later?

Comment (parhamsaremi) [reply commitlint/plugins.test.ts:487]: That was the best way I could think of. We are sure that there is an error we just want to check if the message is what we expect. Do you have any suggestions?

Comment (aarani) [reply commitlint/plugins.test.ts:487]: > That was the best way I could think of. We are sure that there is an error we just want to check if the message is what we expect. Do you have any suggestions?

true, error always happens here so not really, I don't have a suggestion, I guess this is our only option then.

Comment (parhamsaremi) [reply commitlint/plugins.test.ts:487]: I agree that it's not ideal and EOL was the shortest thing we could think of that may never change in the error message :sweat_smile:
